src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2255,21 +2335,17 @@ internal BoundExpression BindBooleanExpression(ExpressionSyntax node, Diagnostic
                             Error(diagnostics, ErrorCode.WRN_IncorrectBooleanAssg, assignment.Syntax);
                         }
                     }
-
-                    return expr;
```

Comment:
Why do we need to create a conversion in the identity case?

Reply:
Because it could be a switch expression, in which case we need to push the conversion to `bool` down into each branch.

---
In reply to: [301342606](https://github.com/dotnet/roslyn/pull/37052#discussion_r301342606) [](ancestors = 301342606)